lib/dradis/plugins/appscan/importer.rb

```diff
@@ -0,0 +1,117 @@
+module Dradis::Plugins::Appscan
+  class Importer < Dradis::Plugins::Upload::Importer
+    # The framework will call this function if the user selects this plugin from
+    # the dropdown list and uploads a file.
+    # @returns true if the operation was successful, false otherwise
+    def import(params = {})
+      logger.info { "Parsing Appscan Source output from #{params[:file]}..." }
+      doc = Nokogiri::XML(File.read(params[:file]))
+      logger.info { 'Done.' }
+
+      @assessment_run = doc.xpath('/AssessmentRun').first
+
+      logger.info { 'Validating Appscan Source output...' }
+      unless @assessment_run &&
+             @assessment_run.attribute('version') &&
+             @assessment_run.attribute('version').text =~ /^9\./
+        error = {
+          'Title' => 'Invalid file format',
+          'File name' => File.basename(params[:file]),
+          'Description' => "The file you uploaded doesn't seem to be a valid \
+                            .ozasmt file."
+        }
+        logger.fatal { error['Description'] }
+        error = error.map { |k, v| format("#[%s]#\n%s\n", k, v) }.join("\n\n")
+        content_service.create_note text: error
+        return false
+      end
+      logger.info { 'Done.' }
+
+      logger.info { 'Processing report...' }
+      @issues = {}
+      @node   = content_service.create_node(label: 'Appscan Source')
+
+      asmnt_files = @assessment_run.xpath('Assessment/Assessment/AsmntFile')
+      asmnt_files.each do |asmnt_file|
+        findings = asmnt_file.xpath('Finding')
+        already_processed = []
+        findings.each do |finding|
```

Comment:
Instead of saving the processed finding_id's and skipping the duplicates, you can use the Array method `uniq` to filter the duplicate `data_id`s. Ex: `findings.uniq { |f| f[:data_id] }.each`. Not sure how this would work, though, if you have a specific duplicate you want to keep.